src/circuits/blake2s/blake2s_setup.tcc

```diff
@@ -0,0 +1,303 @@
+#ifndef __ZETH_BLAKE2S_SETUP_TCC__
```

Comment:
There does not seem to be the corresponding `hpp` file.
Please add the `.hpp` file to be consistent with the rest of the code, include this tcc file after closing the `libzeth` namespace and add reference paper in the comments at the beginning of the file.

Reply:
The functions are alreadu defined in blake2s_comp.hpp . I separated these functions (setup & computation) for clarity (since the file would have been very long). Do you still want an .hpp ?

I could also rename this file as blake2s_comp.cpp as it was done for /zeth/src/types/bits.cpp .

Reply:
> I could also rename this file as blake2s_comp.cpp as it was done for /zeth/src/types/bits.cpp

Let's keep the `.tcc` extension as this is a templatized file.

> The functions are alreadu defined in blake2s_comp.hpp . I separated these functions (setup & computation) for clarity (since the file would have been very long). Do you still want an .hpp

Then maybe rename the file `blake2s_comp_setup.tcc` so we know it's part of the `comp` group of files